packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-checkbox-group/gux-form-field-checkbox-group.service.ts

```diff
@@ -0,0 +1,52 @@
+import simulateNativeEvent from '@utils/dom/simulate-native-event';
+
+export function setAllCheckboxInputs(
+  root: HTMLElement,
+  checked: boolean
+): void {
+  getCheckboxInputs(root).forEach(checkboxInput => {
+    if (checkboxInput.checked !== checked && !checkboxInput.disabled) {
+      checkboxInput.checked = checked;
+      simulateNativeEvent(checkboxInput, 'input');
+      simulateNativeEvent(checkboxInput, 'change');
+    }
+  });
+}
+
+export function setMainCheckboxElementCheckedState(
+  root: HTMLElement,
+  mainCheckboxElement: HTMLInputElement
+): void {
+  if (mainCheckboxElement) {
+    const { count, total } = getSelectedColumnCount(root);
+    if (count === 0) {
+      mainCheckboxElement.indeterminate = false;
+      mainCheckboxElement.checked = false;
+    } else if (count === total) {
+      mainCheckboxElement.indeterminate = false;
+      mainCheckboxElement.checked = true;
+    } else {
```

Comment:
I dont think this logic currently works as,

`getCheckboxInputs` returns all the checkbox inputs in the DOM including the parent checkbox which I think should be excluded for this branch to execute.